From 033e0f05b592bc315c8278fce37f100d90bc64b4 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Wed, 18 Jul 2018 09:31:17 +0100
Subject: [PATCH 360/454] lan78xx: Fix link status notifications

The patch to allow packet reception before the first transmission broke
the notification and handling of link status changes. Move the new call
to tasklet_schedule into lan78xx_link_reset to fix it.

See: https://github.com/raspberrypi/linux/issues/2617

Fixes: d407fc229cdc ("net: lan78xx: fix rx handling before first packet is send")
Suggested-by: Stefan Wahren <stefan.wahren@i2se.com>
Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 drivers/net/usb/lan78xx.c | 2 --
 1 file changed, 2 deletions(-)

--- a/drivers/net/usb/lan78xx.c
+++ b/drivers/net/usb/lan78xx.c
@@ -2585,8 +2585,6 @@ static int lan78xx_open(struct net_devic
 
 	dev->link_on = false;
 
-	tasklet_schedule(&dev->bh);
-
 	lan78xx_defer_kevent(dev, EVENT_LINK_RESET);
 done:
 	usb_autopm_put_interface(dev->intf);
